lib/ui/views/refund_blockchain_selection/bloc/provider.dart

```diff
@@ -1,39 +1,70 @@
 /// SPDX-License-Identifier: AGPL-3.0-or-later
+import 'package:aebridge/application/session/provider.dart';
+import 'package:aebridge/domain/models/bridge_blockchain_environment.dart';
 import 'package:aebridge/ui/views/refund/bloc/provider.dart';
 import 'package:aebridge/ui/views/refund_blockchain_selection/bloc/state.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:archethic_dapp_framework_flutter/archethic_dapp_framework_flutter.dart'
+    as aedappfm;
+import 'package:archethic_wallet_client/archethic_wallet_client.dart' as awc;
+import 'package:riverpod_annotation/riverpod_annotation.dart';
 
-final _blockchainSelectionFormProvider = NotifierProvider.autoDispose<
-    BlockchainSelectionFormNotifier, RefundBlockchainSelectionFormState>(
-  () {
-    return BlockchainSelectionFormNotifier();
-  },
-  name: 'BlockchainSelectionFormNotifierProvider',
-);
+part 'provider.g.dart';
 
+@riverpod
 class BlockchainSelectionFormNotifier
-    extends AutoDisposeNotifier<RefundBlockchainSelectionFormState> {
-  BlockchainSelectionFormNotifier();
-
+    extends _$BlockchainSelectionFormNotifier {
   @override
-  RefundBlockchainSelectionFormState build() {
-    final isTestnetSelected = ref.watch(
-      refundFormNotifierProvider.select((refund) => refund.isTestnetSelected),
+  Future<RefundBlockchainSelectionFormState> build() async {
+    var isTestnetSelected = ref.watch(
+      refundFormNotifierProvider.select((bridge) => bridge.isTestnetSelected),
     );
+
+    var isTestnetIncludedComponentDisplayed = true;
```

Comment:
We could factorize this between `bridge_blockchain_selection`.

I suggest creating a common mixin.

``` dart title="blockchain_selection_mixin.dart"
mixin BlockchainSelectionMixin {
   Future<BridgeBlockchainEnvironment?> get selectedArchethicEnvironment {
      final archethicDAppClient = await aedappfm.sl.getAsync<awc.ArchethicDAppClient>();
      final archethicEndpoint = await archethicDAppClient.getEndpoint().valueOrNull();
     
     return switch(archethicEndpoint) {
        kArchethicEndPointMainnet => BridgeBlockchainEnvironment.mainnet,
        kArchethicEndPointTestnet => BridgeBlockchainEnvironment.testnet,
        _ => BridgeBlockchainEnvironment.devnet
        
     };
}
```

```dart title="refund_blockchain_selection.dart"
// ...
  @override
  Future<RefundBlockchainSelectionFormState> build() async {
    var isTestnetSelected = ref.watch(
      refundFormNotifierProvider.select((bridge) => bridge.isTestnetSelected),
    );

    final archethicEnvironment = await selectedArchethicEnvironment;

    return RefundBlockchainSelectionFormState(
      testnetIncluded: archethicEnvironment != BridgeBlockchainEnvironment.mainnet,
      archethicEnvironment: archethicEnvironment,
      isTestnetIncludedComponentDisplayed: archethicEnvironment == null,
    );
  }
// ...
```

Reply:
This shows that the  properties `testnetIncluded` and `isTestnetIncludedComponentDisplayed` are redundant.

Those properties should be replaced by getters in `RefundBlockchainSelectionFormState`.